File: Python/bit-operation/reverse_bits.py
def reverseBits(n: int) -> int:
    res = 0
    for i in range(32):
        if n == 0: break
        res |= (n & 1) << (31 - i)
        n = n >> 1
    
    return res


if __name__ == "__main__":
    print(reverseBits(-3))

File: core/tmp/Java/src/com/dbc/algorithm/BitOperation/ReverseBits.java
package com.dbc.algorithm.BitOperation;

public class ReverseBits {
    public int reverseBits(int n) {
        int res = 0;
        for (int i = 0; i < 32 && n != 0; i++) {
            res |= (n & 1) << (31 - i);
            n = n >> 1;
        }
        return res;
    }
}
